third_party/amd/lib/TritonAMDGPUToLLVM/OptimizeLDSUtility.cpp

```diff
@@ -0,0 +1,133 @@
+#include "OptimizeLDSUtility.h"
+#include "TritonAMDGPUToLLVM/Passes.h"
+#include "mlir/Pass/Pass.h"
+#include "triton/Analysis/Allocation.h"
+#include "triton/Analysis/Utility.h"
+#include "triton/Conversion/TritonGPUToLLVM/Patterns.h"
+#include "triton/Dialect/Triton/IR/Dialect.h"
+#include "triton/Dialect/TritonGPU/IR/Attributes.h"
+#include "triton/Dialect/TritonGPU/IR/Dialect.h"
+#include <numeric>
+
+using namespace mlir;
+
+namespace mlir {
+namespace triton {
+namespace AMD {
+
+constexpr int kPtrBitWidth = 64;
```

Comment:
Do we really need to hardcode the pointer bitwidth? Can we just use inline constant?

Reply:
This part is copied from `Allocation.cpp` (it is not part of public interface).
Maybe I can actually take this part in some public interface, for example in `Analysis/Utility` module.

Reply:
This is what I was talking about: https://github.com/binarman/triton/pull/6